src/fs.rs

```diff
@@ -387,16 +403,40 @@ impl<IO: Read + Write + Seek, TP, OCC> FileSystem<IO, TP, OCC> {
             FsInfoSector::default()
         };
 
+        let mut bpb_status_flags = bpb.status_flags();
+
         // if dirty flag is set completly ignore free_cluster_count in FSInfo
-        if bpb.status_flags().dirty {
+        if bpb_status_flags.dirty() {
+            if options.ignore_dirty_flag {
+                warn!("BPB is dirty, clearing dirty flag.");
```

Comment:
it's not BPB being dirty, it's entire filesystem. Please change it, e.g. "Filesystem is dirty (BPB), clearing..."